datafusion/optimizer/src/eliminate_cross_join.rs

```diff
@@ -96,6 +96,7 @@ impl OptimizerRule for EliminateCrossJoin {
                 filter.input.as_ref(),
                 LogicalPlan::Join(Join {
                     join_type: JoinType::Inner,
+                    null_equals_null: false,
```

Comment:
This rule also has it's own logic for identifying join keys in `extract_possible_join_keys` function. Shouldn't `is not distinct from` operator be supported there (perhaps by reusing logic from `extract_equijoin_predicated_rule`)?